pkg/adapter/incoming.go

```diff
@@ -86,6 +106,79 @@ func (l *listener) detectIncoming(ctx context.Context, req *http.Request, payloa
 	return true, repo, nil
 }
 
+func (l *listener) getAndUpdateInstallationID(ctx context.Context, req *http.Request, run *params.Run, repo *v1alpha1.Repository) error {
+	jwtToken, err := GenerateJWT(ctx, run)
+	if err != nil {
+		return err
+	}
+
+	installationURL := keys.APIURL + keys.InstallationURL
+	enterpriseURL := req.Header.Get("X-GitHub-Enterprise-Host")
+	if enterpriseURL != "" {
+		installationURL = enterpriseURL + keys.InstallationURL
+	}
+
+	res, err := getResponse(ctx, http.MethodGet, installationURL, jwtToken, run)
+	if err != nil {
+		return err
+	}
+
+	defer res.Body.Close()
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+
+	installationData := []gt.Installation{}
+	if err = json.Unmarshal(data, &installationData); err != nil {
+		return err
+	}
+
+	/* each installationID can have list of repository
+	ref: https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-an-installation ,
+	     https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#list-repositories-accessible-to-the-app-installation */
+	for i := range installationData {
+		l.event.Provider.URL = enterpriseURL
+		gh := github.New()
+		if *installationData[i].ID != 0 {
```

Comment:
if the `ID` is `nil`, leads panic

Reply:
:thinking: 
I am accessing *installationData[i].ID when there is installationData 
So i did not face this panic issue 
will it lead panic ?

Reply:
`ID` is a pointer, can be `nil`
https://go.dev/play/p/Rirp8FkYtNh

Reply:
Oh ya there can be panic issue
As installation ID was never nil so did not thought about that

But added condition to make sure that scenario